internal/guest/runtime/hcsv2/container.go

```diff
@@ -181,3 +185,22 @@ func (c *Container) GetStats(ctx context.Context) (*v1.Metrics, error) {
 func (c *Container) modifyContainerConstraints(ctx context.Context, rt prot.ModifyRequestType, cc *prot.ContainerConstraintsV2) (err error) {
 	return c.Update(ctx, cc.Linux)
 }
+
+// adjustExecSpec adjusts an OCI runtime specs process field provided for an execed process to contain some of the settings set on the containers spec.
+// Some of the OCI spec settings are configured in the guest as there's not enough information available on the host to set them, so the spec set for the
+// exec may be missing some things that were present in the final version of the containers spec.
+func adjustExecSpec(containerSpec *oci.Process, execSpec *oci.Process) {
+	// One of the aforementioned settings is the containers user configured for the init process as we can't verify that the user exists in
+	// the container image on the host. A consequence of the user example is any exec will run as root instead of the user configured for
+	// the container which can be problematic.
+	//
+	// Currently through cri there's no way to set a user for an exec, so just assigning whatever user the container ran as to the execs spec
```

Comment:
This looks good. I also like the idea of checking if uid/gid is changed at all. Because even though this doesn't allow a user to exec as root, I think running as the container user should still provide almost all the permissions required to do whatever operations the exec is trying to do.

However, I feel we can avoid talking about cri in here. This can be rewritten to something like `There is no way to tell if a client supplied root as the user for an exec explicitly or if they simply didn't fill in the spec, as the default values of uid 0 and gid 0 are also the uid & gid values of the root user. To try and cover as many use cases as possible, we check if the uid/gid are changed at all and if not we just copy whatever user was set in the container spec to the exec spec.` What do you think?

Reply:
Sure, we can take off the cri bit. Changing!